.github/workflows/update-ssh-verification-docker.yml

```diff
@@ -0,0 +1,52 @@
+
+name: Update SSH Verification Docker on file change
+
+on:
+  push:
+    paths:
+      - trusted-fingerprint/docker/app.py
+      - trusted-fingerprint/docker/Dockerfile
+      - .github/workflows/update-ssh-verification-docker.yml
+    branches: [main]
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  update-lambda:
+    runs-on: "ubuntu-22.04"
+    environment: "Prod"
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: configure-aws-credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{secrets.AWS_SSM_SSH_ROLE_ARN}}
+          role-session-name: UpdateTrustedFingerprintDockerSession
+          aws-region: ${{secrets.TRUSTED_FINGERPRINT_SERVER_REGION}}
+
+      - name: setup ssh
+        run: |
+          HOMEDIR=$(eval echo ~)
+          mkdir -p "$HOMEDIR/.ssh"
+          cd "$HOMEDIR/.ssh/"
+          ssh-keygen -q -t ed25519 -N '' -f $HOMEDIR/.ssh/id_ed25519
+          cat <<EOF | tee $HOMEDIR/.ssh/config
+          Host ${{secrets.TRUSTED_FINGERPRINT_INSTANCE_ID}}
+            ProxyCommand aws ssm start-session --target %h --document-name AWS-StartSSHSession --parameters 'portNumber=%p'
+          EOF
+
+      - name: Deploy to trusted fingerprint server
+        run: |
+          HOMEDIR=$(eval echo ~)
+          aws ec2-instance-connect send-ssh-public-key \
+             --instance-os-user "deployuser" \
+             --instance-id "${{secrets.TRUSTED_FINGERPRINT_INSTANCE_ID}}" \
+             --ssh-public-key "$(cat $HOMEDIR/.ssh/id_ed25519.pub)"
+          (cd ./trusted-fingerprint/docker && zip -rq trusted-fingerprint.zip ./* && mv ./trusted-fingerprint.zip ../..)
+          scp ./trusted-fingerprint.zip ${{secrets.TRUSTED_FINGERPRINT_INSTANCE_ID}}:/opt/server/trusted-fingerprint
+          ssh -T deployuser@${{secrets.TRUSTED_FINGERPRINT_INSTANCE_ID}} "bash run.sh" && echo "Deployment success"
```

Comment:
run.sh is not in home directory of deployuser, you will need to navigate to the right directory.